portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```
@@ -29,25 +30,70 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.text.ParseException;
+import java.util.TimeZone;
+import java.util.Calendar;
+import java.text.DateFormat;
 
 @WebServlet("/comment-section")
 public class DataServlet extends HttpServlet {
 
+  Integer defaultCommentsNumber = 4;
+  SimpleDateFormat DateFor = new SimpleDateFormat("dd/MM/yy kk:mm:ss");
+  TimeZone timeZone = TimeZone.getTimeZone("Europe/Bucharest");
+    
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     Query query = new Query("Comment");
+    String commentsOrder;
+    try {
+      commentsOrder = request.getParameter("commentsorder");
+      if (commentsOrder.equals("DESC")) {
+        query.addSort("timestamp", SortDirection.DESCENDING);
+      } else {
+        query.addSort("timestamp", SortDirection.ASCENDING);
+      }
+    } catch (Exception e) {
+      //No sorting.
+    }
 
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
 
+     /**
+    * Limit the number of comments to a value chosen by the user.
+    * Extract that value from a query string.
+    * If it does not exist, show 4 comments.
+    */
+
+    Integer commentsNumber;
+    Integer commentsCount = 0;
+
+    try {
+      commentsNumber = Integer.parseInt(request.getParameter("commentsnumber"));
+      if (commentsNumber == 1) {
+          commentsNumber = results.countEntities();
+      }
+    } catch (Exception e) {
```

The only function which can raise an error here is parseInt I think (could be wrong).

This function is documented to throw a NumberFormatException. Use more specific error messages where possible and try to keep try/catch blocks to a minimal size.
Generally 'catch all exceptions' code is a code smell, you probably never want to do that really (sometimes you do, but it's not really common).

---

Alright, will change that